Sources/TuistDependencies/Mappers/ExternalDependencyPathWorkspaceMapper.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+
+public final class ExternalDependencyPathWorkspaceMapper: WorkspaceMapping {
+    public init() {}
+
+    public func map(workspace: WorkspaceWithProjects) throws -> (WorkspaceWithProjects, [SideEffectDescriptor]) {
+        var workspace = workspace
+        let mappedProjects = try workspace.projects.map(map(project:))
+        workspace.projects = mappedProjects.map(\.0)
+        return (
+            workspace,
+            mappedProjects.flatMap(\.1)
+        )
+    }
+
+    // MARK: - Helpers
+
+    private func map(project: Project) throws -> (Project, [SideEffectDescriptor]) {
+        guard project.isExternal,
+              // We don't want to update local packages (which are defined outside the `checkouts` directory in `.build`
+              project.path.parentDirectory.parentDirectory.basename == Constants.SwiftPackageManager.packageBuildDirectoryName
```

Comment:
We are computing the path from the folder where the directory is located, right?

Wouldn't be simpler (and more future proof) to calculate the path from the Tuist root (i.e. where the `Tuist` folder is located)?
I would put things in `Tuist/Dependencies/Derived/<Project>/<Target>` or something similar

Reply:
that would be another dir that users need to gitignore, unnecessarily so. I don't think the inner structure of .build will change and I think it's good to have the derived directory close to the package sources.

Reply:
They should git ignore the whole Tuist/Dependencies already, isn't it?

Reply:
But that directory doesn't exist anymore from tuist 4

Reply:
And I don't get what's the benefit of creating another directory when `.build` should already be gitignored and contains everything related to the SPM dependencies 🤷‍♂️